python/GafferSceneUI/PrimitiveInspector.py

```diff
@@ -0,0 +1,321 @@
+##########################################################################
+#
+#  Copyright (c) 2018, Don Boogert. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of Don Boogert nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import Gaffer
+import GafferScene
+
+import IECore
+import IECoreScene
+
+import GafferUI
+import GafferSceneUI
+
+import collections
+
+
+def _getPrimvarToolTip( primvar ):
+	toolTip = ""
+
+	_type = type( primvar.data )
+
+	if primvar.indices:
+		toolTip += "indexed "
+
+	if IECore.isSequenceDataType( primvar.data ):
+		_type = IECore.valueTypeFromSequenceType( _type )
+		toolTip += "array: "
+
+	toolTip += _type.__name__
+
+	try:
+		toolTip += " (" + str( primvar.data.getInterpretation() ) + ")"
+	except:
+		pass
+
+	return toolTip
+
+
+def _orderPrimitiveVariables( primVarNames ) :
+	priorityOrder = ['P', 'N', 'uv', 'Pref']
+	result = []
+
+	for priorityPrimVar, i in enumerate( priorityOrder ) :
+		if priorityPrimVar in primVarNames :
+			result.append( priorityPrimVar )
+			del primVarNames[i]
+
+	result.extend( sorted( primVarNames ) )
+
+	return result
+
+def conditionPrimvar( primvar ) :
+
+	scalarToVector = {
```

Comment:
I think something like this might be a shorter alternative :

```
if hasattr( data, "value" ) :
    data = ECore.DataTraits.dataFromElement( [ data.value ] )
```